profiler/src/main/java/com/splunk/opentelemetry/profiler/BasicJfrRecordingFile.java

```diff
@@ -44,15 +48,25 @@ public boolean tryAdvance(Consumer<? super RecordedEvent> action) {
               action.accept(jfr.readEvent(file, path));
               return true;
             }
+            closeSafely(file);
             return false;
           }
 
           public void forEachRemaining(Consumer<? super RecordedEvent> action) {
             while (file.hasMoreEvents()) {
               action.accept(jfr.readEvent(file, path));
             }
+            closeSafely(file);
           }
         },
         false);
   }
+
+  private void closeSafely(RecordingFile file) {
+    try {
+      file.close();
+    } catch (IOException e) {
+      logger.warn("Error closing JFR file", e);
```

Comment:
Will an end-user see this WARN log? What should they do about it? Because they will be alarmed by that.

Reply:
If they look at the logs and have it configured for WARN or below, then yes.  What should they do about it? I dunno. It could help inform when looking into other problems? There are some wild failure modes that cause close to fail. Would you prefer that we silently eat it?

Reply:
For me "WARN" always means "not a problem right now but you should take a look and fix it anyway". User cannot do anything to actually remedy this WARN, can they?

Reply:
For me, WARN usually means "Hey something kinda bad happened that you probably should know about, but your app didn't necessarily hit an error".  If their filesystem got turned read only, for example, they could change it back to read/write to fix the problem.  So I I dunno, I think maybe yeah they can do something....but are you suggesting we make it a lower level then?

Reply:
You convinced me :)